core/test/config/type_descriptor.cpp

```diff
@@ -21,27 +21,40 @@ TEST(TypeDescriptor, TemplateCreate)
 
         ASSERT_EQ(td.get_value_typestr(), "float64");
         ASSERT_EQ(td.get_index_typestr(), "int32");
+        ASSERT_EQ(td.get_global_index_typestr(), "int64");
+    }
+    {
+        SCOPED_TRACE("specify global indextype");
+        auto td = make_type_descriptor<float, int, int>();
+
+        ASSERT_EQ(td.get_value_typestr(), "float32");
+        ASSERT_EQ(td.get_index_typestr(), "int32");
+        ASSERT_EQ(td.get_global_index_typestr(), "int");
```

Comment:
Maybe to be clear:
```suggestion
        ASSERT_EQ(td.get_global_index_typestr(), "int32");
```

Reply:
interesting... "int" should be wrong but the test seems not to catch it

Reply:
I miss adding it in the CMakeLists..